pkg/tools/multiexecutor/utils.go

```diff
@@ -14,15 +14,22 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package serialize
+// Package multiexecutor provides a structure MultiExecutor that can be used to guarantee exclusive by ID, in order execution of functions.
+package multiexecutor
 
 import (
 	"sync"
 
 	"github.com/edwarnicke/serialize"
 )
 
-type multiExecutor struct {
+// Executor is a serialize.Executor interface type
+type Executor interface {
```

Comment:
I am not sure that this should be removed here from `serialize`, because almost no one who takes `Executor` from context knows anything about `MultiExecutor` but something about `serialize`.

Reply:
Removed serialize from comment. All "context" methods already left at serialize

Reply:
This interface required by MultiExecutor to pass all AsyncExec calls through it to be sure that executor will be deleted in time. 
All context methods are left at serialize

Reply:
`MultiExecutor` is using `Executor` as a return value mostly to make code cleaner than it actually needs it. So it is not an any strong consumer, it is only a producer.

Reply:
Yes, but it will create possible sittuation, when two Executors are exists for the same ID